src/main/java/com/facebook/openwifirrm/ucentral/UCentralClient.java

```diff
@@ -614,6 +624,39 @@ public void setServiceEndpoint(String service, ServiceEvent event) {
 		}
 	}
 
+	/**
+	 * Validate the token is still valid.
+	 */
+	public boolean validateToken(String token) {
+		if (!tokenCache.containsKey(token)) {
```

Comment:
nit: just to avoid the double-lookup (with line 655), do:
```
Long expiry = tokenCache.get(token);
if (expiry == null) {
  // ...
  expiry = result.tokenInfo.created + result.tokenInfo.expires_in;
  tokenCache.put(token, expiry);
}
```